### PR TITLE
Rename various CMR structs and methods

### PR DESCRIPTION
**Pull request by wallyworld**
One of a few PR's to fix the underlying CMR data model and properly name various structs and methods.

QA: bootstrap to ensure no repressions. Mechanical change only behind a feature flag.

### CODE REVIEW

File: api/watcher/watcher.go [hunk 1]
```diff
@@ -515,24 +515,24 @@ func (w *migrationStatusWatcher) Changes() <-chan watcher.MigrationStatus {
...
 	commonWatcher
 	caller                        base.APICaller
 	applicationRelationsWatcherId string
-	out                           chan watcher.ApplicationRelationsChange
+	out                           chan watcher.RemoteRelationsChange
 }
 
-// NewApplicationRelationsWatcher returns an ApplicationRelationsWatcher which
-// communicates with the ApplicationRelationsWatcher API facade to watch
+// NewRemoteRelationsWatcher returns a RemoteRemoteRelationsWatcher which
```
Comment (axw) [inline api/watcher/watcher.go]: s/RemoteRemote/Remote/

File: api/watcher/watcher.go [hunk 2]
```diff
@@ -515,24 +515,24 @@ func (w *migrationStatusWatcher) Changes() <-chan watcher.MigrationStatus {
...
+// communicates with the RemoteRemoteRelationsWatcher API facade to watch
```
Comment (axw) [inline api/watcher/watcher.go]: ditto

File: api/watcher/watcher.go [hunk 3]
```diff
@@ -515,24 +515,24 @@ func (w *migrationStatusWatcher) Changes() <-chan watcher.MigrationStatus {
...
-// is involved in, including changes to the units involved in those
-// relations, and their settings.
-type applicationRelationsWatcher struct {
+// remoteRelationsWatcher will sends changes to relations an application
+// is involved in with another remote application, including changes to the
+// remote units involved in those relations, and their settings.
+type remoteRelationsWatcher struct {
 	commonWatcher
 	caller                        base.APICaller
 	applicationRelationsWatcherId string
```
Comment (axw) [inline api/watcher/watcher.go]: rename field?

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**